### PR TITLE
Attention, Tunnel Env, Bug Fixes

### PR DESCRIPTION
**Pull request by SumeetBatra**
- General get_obs_rel function that includes attention 
- Bug fix where room dims wouldn't change during evaluation time (callback func allows the env to update its room_dims attr) 
- Tunnel scenario (not included in mixed scenario for now, but can if we want to down the line)

### CODE REVIEW

File: gym_art/quadrotor_multi/quadrotor_single.py [hunk 1]
```diff
@@ -947,8 +947,8 @@ def make_observation_space(self):
         self.obs_space_low_high = {
             "xyz": [-(self.room_box[1] - self.room_box[0]), self.room_box[1] - self.room_box[0]],
             "xyzr": [-(self.room_box[1] - self.room_box[0]), self.room_box[1] - self.room_box[0]],
-            "vxyz": [-self.dynamics.vxyz_max * np.ones(3), self.dynamics.vxyz_max * np.ones(3)],
-            "vxyzr": [-self.dynamics.vxyz_max * np.ones(3), self.dynamics.vxyz_max * np.ones(3)],
+            "vxyz": [-2*self.dynamics.vxyz_max * np.ones(3), 2*self.dynamics.vxyz_max * np.ones(3)],
+            "vxyzr": [-2*self.dynamics.vxyz_max * np.ones(3), 2*self.dynamics.vxyz_max * np.ones(3)],
```
Comment (Zhehui-Huang) [inline gym_art/quadrotor_multi/quadrotor_single.py]: I don't think we should change vxyz and vxyzr. They are not the relative velocity

**Review by Zhehui-Huang — CHANGES_REQUESTED** [review]
I basically okay with your changes. My only question is that have you tested your changes? Sometimes, it's not easy to find the bugs until we run some experiments.

File: gym_art/quadrotor_multi/quadrotor_single.py (hunk 1 above)
Comment (SumeetBatra) [reply gym_art/quadrotor_multi/quadrotor_single.py:951]: We use them to do clipping on relative velocity

Comment (SumeetBatra) [reply gym_art/quadrotor_multi/quadrotor_single.py]: Whoops, I think I misread the code. I'll change this back

**Comment by SumeetBatra** [comment]
> I basically okay with your changes. My only question is that have you tested your changes? Sometimes, it's not easy to find the bugs until we run some experiments.

All of these have been tested locally with train and evaluation scripts. I've started a full run on the server but it hasn't finished yet

File: gym_art/quadrotor_multi/quadrotor_multi.py [hunk 2]
```diff
@@ -93,6 +93,8 @@ def __init__(self,
 
         if self.swarm_obs == 'pos_vel':
             self.neighbor_obs_size = 6
+        elif self.swarm_obs == 'attn':
```
Comment (alex-petrenko) [inline gym_art/quadrotor_multi/quadrotor_multi.py:97]: This name should be explicit and descriptive

File: gym_art/quadrotor_multi/quadrotor_single.py [hunk 3]
```diff
@@ -947,8 +947,8 @@ def make_observation_space(self):
         self.obs_space_low_high = {
             "xyz": [-(self.room_box[1] - self.room_box[0]), self.room_box[1] - self.room_box[0]],
             "xyzr": [-(self.room_box[1] - self.room_box[0]), self.room_box[1] - self.room_box[0]],
-            "vxyz": [-self.dynamics.vxyz_max * np.ones(3), self.dynamics.vxyz_max * np.ones(3)],
-            "vxyzr": [-self.dynamics.vxyz_max * np.ones(3), self.dynamics.vxyz_max * np.ones(3)],
+            "vxyz": [self.dynamics.vxyz_max * np.ones(3), self.dynamics.vxyz_max * np.ones(3)],
```
Comment (alex-petrenko) [inline gym_art/quadrotor_multi/quadrotor_single.py]: bug

**Review by alex-petrenko — APPROVED** [review]

**Pull request merged**